DMCompiler/Compiler/DM/DMParser.cs

```diff
@@ -979,9 +979,9 @@ public DMASTProcStatement ProcStatement()
                     Newline();
 
 
-                    //Implicit "in world"
+                    //Implicit "in world", handled later
                     if (variableDeclaration.Value is null && rangeEnd is null && defaultStep) {
-                        return new DMASTProcStatementForList(loc, initializer, variable, new DMASTIdentifier(loc, "world"), GetForBody());
+                        return new DMASTProcStatementForList(loc, initializer, variable, null, GetForBody());
```

Comment:
I think it would be better to have a new AST node for this, something like `DMASTProcStatementForType`